app/assets/javascripts/components/DisplayLogicFormField.jsx

```diff
@@ -0,0 +1,56 @@
+class DisplayLogicFormField extends React.Component {
+  constructor(props) {
+    super();
+    this.state = props;
+    this.changeDisplayOption = this.changeDisplayOption.bind(this);
+    this.buildConditions = this.buildConditions.bind(this);
+    this.addCondition = this.addCondition.bind(this)
+  }
+
+  changeDisplayOption(event) {
+    let value = event.target.value
+
+    this.setState({display_if: value})
+  }
+
+  buildConditions(args) {
+    if(this.state.display_if != "always") {
+      return (
+        <div>
+          {this.state.display_conditions.map((props, index) => <ConditionsFormField key={index} {...props}/>)}
+          <button onClick={this.addCondition} type="button" className="btn add_condition">
+            <i className="fa fa-plus"></i> {I18n.t("form_item.add_condition")}
+          </button>
+        </div>
+      )
+    }
+  }
+
+  addCondition() {
+    let conditions = this.state.display_conditions
+    conditions.push({refable_qings: this.state.refable_qings, operator_options: [], questioning_id: this.state.id})
+
+    this.setState({display_conditions: conditions})
+  }
+
+  render() {
+    let select_props = {
```

Comment:
The similar vars in `ConditionsFormField` aren't.

Reply:
Ok I guess we need a style guide. It's tough because there is so much stuff coming from the API that is underscore style. But the JS world is pretty heavily camel case. What do you think?

Reply:
I'd prefer we stick to `underscore_style` for data, personally. We can use `camelCase` for functions, but anything we serialize from ruby should match ruby's representation, that includes our props if not the variables we store those props in, which I could see going either way. Personally, I'd rather ignore the JS ecosystem's conventions on this because I think they're wrong, and underscore style in the react code better matches our coffeescript stuff, which is going to coexist in the codebase with react longer than our pure/legacy javascript will.